Services/CO.CDP.Organisation.WebApi/UseCase/LookupOrganisationUseCase.cs

```diff
@@ -3,11 +3,58 @@
 using CO.CDP.OrganisationInformation.Persistence;
 
 namespace CO.CDP.Organisation.WebApi.UseCase;
-public class LookupOrganisationUseCase(IOrganisationRepository organisationRepository, IMapper mapper) : IUseCase<string, Model.Organisation?>
+public class LookupOrganisationUseCase(IOrganisationRepository organisationRepository, IMapper mapper) : IUseCase<OrganisationQuery, Model.Organisation?>
 {
-    public async Task<Model.Organisation?> Execute(string name)
+    public async Task<Model.Organisation?> Execute(OrganisationQuery query)
     {
-        return await organisationRepository.FindByName(name)
-            .AndThen(mapper.Map<Model.Organisation>);
+        if (!query.IsValid)
+        {
+            throw new ArgumentException("Organisation name or identifier must be provided.");
```

Comment:
This has to be a more specific (dedicated) exception, so it could be mapped to a 400 error response.

Throwing System exceptions directly, without extending them, should be avoided. The reason for this is such an exception can be thrown by anyone, including third party libraries, and we can't unambiguously map it to a 400 error response.

Reply:
Hi Kuba, Updated this.